airflow/providers/postgres/hooks/postgres.py

```diff
@@ -114,6 +115,25 @@ def get_conn(self) -> connection:
 
         self.conn = psycopg2.connect(**conn_args)
         return self.conn
+    
+    def get_uri(self) -> str:
+        """
+        Extract the URI from the connection.
+
+        :return: the extracted uri.
+        """
+        conn = self.get_connection(getattr(self, self.conn_name_attr))
+        login = ''
+        if conn.login:
+            login = f'{quote_plus(conn.login)}:{quote_plus(conn.password)}@'
+        host = conn.host
+        if conn.port is not None:
+            host += f':{conn.port}'
+        uri = f'{conn.conn_type}://{login}{host}/'
+        schema = self.schema or conn.schema
+        if schema:
+            uri += conn.schema
+        return uri
```

Comment:
`urlunsplit` would save you a lot of trouble :)

Reply:
ok, thanks!